File: 800 - rated/desorting.c
#include<stdio.h>
#include<stdlib.h>
#include<limits.h>
int main(){
    int t,i ,temp = 0;
    long temp1 ;
    scanf("%d",&t);
    while(t--){
        int n;
        temp = 0 ;
        long min = INT_MAX;
        scanf("%d",&n);
        long * nums = (long *)malloc(sizeof(long)* n);
        for(i=0;i<n;i++){
            scanf("%ld",(nums+i));
        }
        for(i=1;i<n;i++){
            if(nums[i] < nums[i-1] ){
                temp = 1;
                break;
            }
            temp1 = (nums[i] - nums[i-1])/2 + 1 ;
            min = (min > temp1) ? temp1 : min ;



        }
        if(temp)
            printf("%d\n",0);
        else 
            printf("%ld\n",min);
        
    }

    return 0;
}

File: 800 - rated/desorting.py
t=int(input())
for _ in range(t):
    n = int(input())
    nums = list(map(int,input().split()))
    m = float("inf")
    temp = False
    for i in range(1,len(nums)):
        if nums[i] < nums[i-1] :
            temp = True
            break
        m = min(m,(nums[i] - nums[i-1])//2  + 1)
    if temp:
        print(0)
    else :
        print(m)
        